superset/views/utils.py

```diff
@@ -176,7 +176,9 @@ def get_datasource_info(
     return datasource_id, datasource_type
 
 
-def apply_display_max_row_limit(sql_results: Dict[str, Any]) -> Dict[str, Any]:
+def apply_display_max_row_limit(
+    sql_results: Dict[str, Any], rows: Optional[int] = None
```

Comment:
is this ever called with `rows` not defined?

Reply:
Yes, synchronous queries will still call this to limit the response:

```
            payload = json.dumps(
                apply_display_max_row_limit(data),
                default=utils.pessimistic_json_iso_dttm_ser,
                ignore_nan=True,
                encoding=None,
            )
```

I think it makes sense to limit the sync response, disallowing users from bypassing it. if the query is returned more than `DISPLAY_MAX_ROWS` and the user needs that data they should run it asynchronously, IMHO.

Reply:
Async vs sync queries are dependent on the config of the datasource right? Which is something the user doesn't have any control over unless they're admin. I guess if you're calling this with an API, then you can decide if you want to make an async or async query yourself though.

I think it's fine, but it's just another little wart with superset that we need to remember

Reply:
Right.